### ISSUE TITLE
RxAndroid dependency runtime crash (NoClassDefFoundError)

### ISSUE THREAD

**Issue by ogero** [germnix/acacia#3]
Hi, 
Im having the following crash when using Acacia in a project without RxAndroid.
Is RxAndroid mandatory on new Acacia 0.1.1 version?
Could you check if PendingInvocation can be resolved near ServiceProxy.java:87 to avoid the exception, or catch it? So Acacia can still be used without adding RxAndroid.

```
07-25 02:54:23.724  23033-23033/com.myapp I/dalvikvm﹕ Failed resolving Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation; interface 5459 'Lrx/Observable$OnSubscribe;'
07-25 02:54:23.728  23033-23033/com.myapp W/dalvikvm﹕ Link of class 'Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation;' failed
07-25 02:54:23.728  23033-23033/com.myapp E/dalvikvm﹕ Could not find class 'com.gmr.acacia.impl.ServiceProxy$PendingInvocation', referenced from method com.gmr.acacia.impl.ServiceProxy.handleDisconnectedInvocation
07-25 02:54:23.728  23033-23033/com.myapp W/dalvikvm﹕ VFY: unable to resolve new-instance 2866 (Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation;) in Lcom/gmr/acacia/impl/ServiceProxy;
07-25 02:54:23.728  23033-23033/com.myapp D/dalvikvm﹕ VFY: replacing opcode 0x22 at 0x0001
07-25 02:54:23.728  23033-23033/com.myapp I/dalvikvm﹕ Failed resolving Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation; interface 5459 'Lrx/Observable$OnSubscribe;'
07-25 02:54:23.728  23033-23033/com.myapp W/dalvikvm﹕ Link of class 'Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation;' failed
07-25 02:54:23.728  23033-23033/com.myapp E/dalvikvm﹕ Could not find class 'com.gmr.acacia.impl.ServiceProxy$PendingInvocation', referenced from method com.gmr.acacia.impl.ServiceProxy.onServiceConnected
07-25 02:54:23.728  23033-23033/com.myapp W/dalvikvm﹕ VFY: unable to resolve check-cast 2866 (Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation;) in Lcom/gmr/acacia/impl/ServiceProxy;
07-25 02:54:23.728  23033-23033/com.myapp D/dalvikvm﹕ VFY: replacing opcode 0x1f at 0x0025
07-25 02:54:23.728  23033-23033/com.myapp I/dalvikvm﹕ Failed resolving Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation; interface 5459 'Lrx/Observable$OnSubscribe;'
07-25 02:54:23.728  23033-23033/com.myapp W/dalvikvm﹕ Link of class 'Lcom/gmr/acacia/impl/ServiceProxy$PendingInvocation;' failed
07-25 02:54:23.728  23033-23033/com.myapp D/dalvikvm﹕ DexOpt: unable to opt direct call 0x684e at 0x03 in Lcom/gmr/acacia/impl/ServiceProxy;.handleDisconnectedInvocation
07-25 02:54:23.804  23033-23033/com.myapp E/dalvikvm﹕ Could not find class 'rx.Observable', referenced from method com.gmr.acacia.AcaciaService.handleObservableOnWorkerThread
07-25 02:54:23.804  23033-23033/com.myapp W/dalvikvm﹕ VFY: unable to resolve check-cast 5460 (Lrx/Observable;) in Lcom/gmr/acacia/AcaciaService;
07-25 02:54:23.804  23033-23033/com.myapp D/dalvikvm﹕ VFY: replacing opcode 0x1f at 0x0006
07-25 02:54:23.804  23033-23033/com.myapp E/dalvikvm﹕ Could not find class 'rx.Observable', referenced from method com.gmr.acacia.AcaciaService.invoke
07-25 02:54:23.804  23033-23033/com.myapp W/dalvikvm﹕ VFY: unable to resolve const-class 5460 (Lrx/Observable;) in Lcom/gmr/acacia/AcaciaService;
07-25 02:54:23.804  23033-23033/com.myapp D/dalvikvm﹕ VFY: replacing opcode 0x1c at 0x001e
07-25 02:54:23.808  23033-23033/com.myapp D/Acacia﹕ Creating SyncService
07-25 02:54:23.880  23033-23033/com.myapp D/Acacia﹕ Service connected.
07-25 02:54:23.892  23033-23033/com.myapp D/AndroidRuntime﹕ Shutting down VM
07-25 02:54:23.892  23033-23052/com.myapp D/Acacia﹕ Starting thread SyncServiceImplThread
07-25 02:54:23.892  23033-23033/com.myapp W/dalvikvm﹕ threadid=1: thread exiting with uncaught exception (group=0xa4cb2b20)
07-25 02:54:23.892  23033-23033/com.myapp E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.myapp, PID: 23033
    java.lang.NoClassDefFoundError: com.gmr.acacia.impl.ServiceProxy$PendingInvocation
            at com.gmr.acacia.impl.ServiceProxy.onServiceConnected(ServiceProxy.java:87)
            at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:1101)
            at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1118)
            at android.os.Handler.handleCallback(Handler.java:733)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:136)
            at android.app.ActivityThread.main(ActivityThread.java:5001)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
            at dalvik.system.NativeStart.main(Native Method)
```

**Comment by ghost** [comment]
For some reason the changes made in develop were not merged correctly into master, so version 0.1.1 didn't include any change.

I released 0.1.2 with the intended changes, so this exception should be gone.

**Comment by ogero** [comment]
I've just updated my dependencies to `com.gmr:acacia:0.1.2`, but the exception is still there:

```
java.lang.NoClassDefFoundError: Failed resolution of: Lrx/Observable;
    at com.gmr.acacia.AcaciaService.invoke(AcaciaService.java:147)
    at com.gmr.acacia.impl.ServiceProxy.handleConnectedInvocation(ServiceProxy.java:161)
    at com.gmr.acacia.impl.ServiceProxy.invoke(ServiceProxy.java:124)
    at java.lang.reflect.Proxy.invoke(Proxy.java:393)
    at $Proxy1.myFancyMethodName(Unknown Source)
```
